purchases/src/main/kotlin/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -227,6 +240,100 @@ internal open class DeviceCache(
 
     // endregion
 
+    // region virtual currencies
+    fun virtualCurrenciesCacheKey(appUserID: String) = "$virtualCurrenciesCacheBaseKey.$appUserID"
+
+    fun virtualCurrenciesLastUpdatedCacheKey(appUserID: String) = "$virtualCurrenciesLastUpdatedCacheBaseKey.$appUserID"
+
+    @Synchronized
+    fun getCachedVirtualCurrencies(appUserID: String): VirtualCurrencies? {
+        return preferences.getString(virtualCurrenciesCacheKey(appUserID), null)
+            ?.let { json ->
+                try {
+                    val cachedJSONObject = JSONObject(json)
+                    return VirtualCurrenciesFactory.buildVirtualCurrencies(body = cachedJSONObject)
+                } catch (e: JSONException) {
+                    null
```

Comment:
Hmm would it be worth adding a log here? (not sure in what situations this may happen.